build/jenkins_e2e.sh

```diff
@@ -61,7 +55,25 @@ golden_nodes=(
 
 max_retries=8
 
+if [[ -a "${CADVISOR_APPLICATION_CREDENTIALS:-}" ]] ; then
+  gcloud auth activate-service-account --key-file="${CADVISOR_APPLICATION_CREDENTIALS}"
+fi
+
+mkdir -p ~/.ssh
+
+if [[ -a "${JENKINS_GCE_SSH_PRIVATE_KEY_FILE:-}" ]] ; then
+  cp ${JENKINS_GCE_SSH_PRIVATE_KEY_FILE} ~/.ssh/google_compute_engine
+fi
+
+if [[ -a "${JENKINS_GCE_SSH_PRIVATE_KEY_FILE:-}" ]] ; then
+  cp ${JENKINS_GCE_SSH_PUBLIC_KEY_FILE} ~/.ssh/google_compute_engine.pub
+fi
+
 ./runner --logtostderr --test-retry-count=$max_retries \
   --test-retry-whitelist=integration/runner/retrywhitelist.txt \
-  --ssh-options "-i /var/lib/jenkins/gce_keys/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no" \
+  --ssh-options "--project=cadvisor-e2e --strict-host-key-checking=no --zone=us-central1-f" \
   ${golden_nodes[*]}
+
+if [[ -a "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]] ; then
```

Comment:
also you want to switch back the service account regardless of test results

Reply:
Any tips on how to do that?

Reply:
hummm, you can store exit value from runner and exit with that in the end, and don't `set -e`?